readme.txt
=== Lightweight Grid Columns ===
Contributors: edge22
Donate link: https://generatepress.com/ongoing-development
Tags: columns, columns shortcode, grid columns
Requires at least: 4.0
Tested up to: 4.5.2
Stable tag: 0.6
License: GPLv2 or later
License URI: http://www.gnu.org/licenses/gpl-2.0.html

Easily add desktop, tablet and mobile friendly columns to your content using an easy to use shortcode.

== Description ==

Lightweight Grid Columns are super easy to use! Install and activate the plugin, then look for the columns icon within your TinyMCE toolbar (see screenshots).

Specify your desired desktop, tablet and mobile widths of the columns, add your content and then insert your shortcode into your content.

Lightweight Grid Columns uses the awesome Unsemantic Framework: (http://unsemantic.com/)

Check out GeneratePress, our awesome WordPress theme! (http://wordpress.org/themes/generatepress)

= Features include: =

* Desktop grid width
* Tablet grid width
* Mobile grid width
* Add custom classes
* Add custom inline styles
* Equal height columns

== Installation ==

There's two ways to install Lightweight Grid Columns.

1. Go to "Plugins > Add New" in your Dashboard and search for: Lightweight Grid Columns
2. Download the .zip from WordPress.org, and upload the folder to the `/wp-content/plugins/` directory via FTP.

In most cases, #1 will work fine and is way easier.

== Frequently Asked Questions ==

= How do I add the shortcodes? =

* Make sure Lightweight Grid Columns is activated.
* While editing your post or page, go into the "Visual Editor".
* Look for the columns icon (see screenshots tab).
* Choose your desktop, tablet and mobile grid widths.
* If you're adding the last column in a row, check the "Last" checkbox (this is important).

= What if I don't want to use the TinyMCE button? =

Fair enough! Simply add the shortcode to your content.

For example, the below will output 4 columns on desktop, 2 columns on tablet and 1 column on mobile

	[lgc_column grid="25" tablet_grid="50" mobile_grid="100"]Some content[/lgc_column]
	
	[lgc_column grid="25" tablet_grid="50" mobile_grid="100"]Some content[/lgc_column]
	
	[lgc_column grid="25" tablet_grid="50" mobile_grid="100"]Some content[/lgc_column]
	
	[lgc_column grid="25" tablet_grid="50" mobile_grid="100" last="true"]Some content[/lgc_column]
	
Take note of our last column - see the last="true" part? Don't forget ;)

= What are my options for percentages? =

This is taken directly from (http://unsemantic.com):

There are grid classes named grid-x where "x" is a number that represents the percentage width of each grid unit. These cover multiples of 5, up to 100 (grid-5, grid-10 ... grid-95, grid-100). There are also classes for dividing a page into thirds: grid-33 and grid-66 which are 33.3333% and 66.6667% wide, respectively. 

= Are there any other options I can use in the shortcode? =

Yes!

You can use:

= class =
	
	[lgc_column grid="25" tablet_grid="50" mobile_grid="100" class="push-25"]Some content[/lgc_column]

= style =

	[lgc_column grid="25" tablet_grid="50" mobile_grid="100" style="padding-left:0px;"]Some content[/lgc_column]
	
= equal_heights =

By default, JS equal heights are set to "true".

    [lgc_column grid="25" tablet_grid="50" mobile_grid="100" equal_heights="true"]Some content[/lgc_column]

== Screenshots ==

1. The columns icon in your TinyMCE editor.
2. The options inside the columns shortcode generator.
3. A look at some awesome columns!

== Changelog ==

= 0.7 =
* Fix weird scroll behavior caused by transitions and matchHeight
* Only load matchHeight on pages where columns are present with it enabled

= 0.6 =
* Fix column class bug causing some columns not to work

= 0.5 =
* Move margin-bottom setting to inner element so style attribute can overwrite it
* Adjust lgc_columns_helper function to fix issue with `<p>` and `<br />` tags around the shortcode
* Add jQuery matchHeight to match the height of columns for better responsive user experience
* Reduce the file size of stylesheet considerably

= 0.4 =
* Show TinyMCE icon on all post types

= 0.3 =
* Fix bug where TinyMCE icon wasn't showing up if SiteOrigin Page Builder and Black Studio TinyMCE Widget were activated together
* Add IE 7 & 8 compatibility

= 0.2 =
* Allow line breaks inside columns

= 0.1 =
* Initial release

== Upgrade Notice ==

= 0.7 =
* Fix weird scroll behavior caused by transitions and matchHeight
* Only load matchHeight on pages where columns are present with it enabled

= 0.6 =
* Fix column class bug causing some columns not to work

= 0.5 =
* Move margin-bottom setting to inner element so style attribute can overwrite it
* Adjust lgc_columns_helper function to fix issue with `<p>` and `<br />` tags around the shortcode
* Add jQuery matchHeight to match the height of columns for better responsive user experience
* Reduce the file size of stylesheet considerably

= 0.4 =
Show TinyMCE icon on all post types

= 0.3 =
Add old IE compatibility

= 0.2 =
Allow line breaks inside your columns

= 0.1 =
* Initial release

changelog.txt
== changelog ==

(6/3/2016) - 0.6
* Fix column class bug causing some columns not to work

----

(6/2/2016) - 0.5
* Move margin-bottom setting to inner element so style attribute can overwrite it
* Adjust lgc_columns_helper function to fix issue with <p> and <br /> tags around the shortcode
* Add jQuery matchHeight to match the height of columns for better responsive user experience
* Reduce the file size of stylesheet considerably

----

(6/20/2015) - 0.4
[*] Show TinyMCE icon on all post types

----

(6/19/2015) - 0.3
[!] Fix bug where TinyMCE icon wasn't showing up if SiteOrigin Page Builder and Black Studio TinyMCE Widget were activated together
[+] Add IE 7 & 8 compatibility

----

(5/21/2015) - 0.2
[*] Allow line breaks inside columns

----

(5/15/2015) - 0.1
Initial Release

js/button.js
(function() {
    tinymce.PluginManager.add('lgc_shortcodes_button', function( editor, url ) {
        editor.addButton( 'lgc_shortcodes_button', {
			title: lgc_add_columns,
            icon: 'icon dashicons-text',
            onclick: function() {
				editor.windowManager.open( {
					title: lgc_columns,
					body: [
					{
						type: 'listbox', 
						name: 'desktop_grid', 
						label: lgc_desktop, 
						'values': [
							{text: '50%', value: '50'},
							{text: '5%', value: '5'},
							{text: '10%', value: '10'},
							{text: '15%', value: '15'},
							{text: '20%', value: '20'},
							{text: '25%', value: '25'},
							{text: '30%', value: '30'},
							{text: '33%', value: '33'},
							{text: '35%', value: '35'},
							{text: '40%', value: '40'},
							{text: '45%', value: '45'},
							{text: '50%', value: '50'},
							{text: '55%', value: '55'},
							{text: '60%', value: '60'},
							{text: '65%', value: '65'},
							{text: '70%', value: '70'},
							{text: '75%', value: '75'},
							{text: '80%', value: '80'},
							{text: '85%', value: '85'},
							{text: '90%', value: '90'},
							{text: '95%', value: '95'},
							{text: '100%', value: '100'}
						]
					},
					{
						type: 'listbox', 
						name: 'tablet_grid', 
						label: lgc_tablet, 
						'values': [
							{text: '50%', value: '50'},
							{text: '5%', value: '5'},
							{text: '10%', value: '10'},
							{text: '15%', value: '15'},
							{text: '20%', value: '20'},
							{text: '25%', value: '25'},
							{text: '30%', value: '30'},
							{text: '33%', value: '33'},
							{text: '35%', value: '35'},
							{text: '40%', value: '40'},
							{text: '45%', value: '45'},
							{text: '50%', value: '50'},
							{text: '55%', value: '55'},
							{text: '60%', value: '60'},
							{text: '65%', value: '65'},
							{text: '70%', value: '70'},
							{text: '75%', value: '75'},
							{text: '80%', value: '80'},
							{text: '85%', value: '85'},
							{text: '90%', value: '90'},
							{text: '95%', value: '95'},
							{text: '100%', value: '100'}
						]
					},
					{
						type: 'listbox', 
						name: 'mobile_grid', 
						label: lgc_mobile, 
						'values': [
							{text: '100%', value: '100'},
							{text: '5%', value: '5'},
							{text: '10%', value: '10'},
							{text: '15%', value: '15'},
							{text: '20%', value: '20'},
							{text: '25%', value: '25'},
							{text: '30%', value: '30'},
							{text: '33%', value: '33'},
							{text: '35%', value: '35'},
							{text: '40%', value: '40'},
							{text: '45%', value: '45'},
							{text: '50%', value: '50'},
							{text: '55%', value: '55'},
							{text: '60%', value: '60'},
							{text: '65%', value: '65'},
							{text: '70%', value: '70'},
							{text: '75%', value: '75'},
							{text: '80%', value: '80'},
							{text: '85%', value: '85'},
							{text: '90%', value: '90'},
							{text: '95%', value: '95'},
							{text: '100%', value: '100'}
						]
					},
					{
						type: 'textbox',
						name: 'content',
						label: lgc_content,
						multiline: true,
						minWidth: 300,
						minHeight: 100,
					},
					{
						type: 'checkbox',
						name: 'last',
						label: lgc_last
					}
					],
					onsubmit: function( e ) {
						editor.insertContent( '[lgc_column grid="' + e.data.desktop_grid + '" tablet_grid="' + e.data.tablet_grid + '" mobile_grid="' + e.data.mobile_grid + '" last="' + e.data.last + '"]' + e.data.content + '[/lgc_column]');
					}
				});
			}
        });
    });
})();

js/jquery.matchHeight.js
/**
* jquery-match-height 0.7.0 by @liabru
* http://brm.io/jquery-match-height/
* License: MIT
*/

;(function(factory) { // eslint-disable-line no-extra-semi
    'use strict';
    if (typeof define === 'function' && define.amd) {
        // AMD
        define(['jquery'], factory);
    } else if (typeof module !== 'undefined' && module.exports) {
        // CommonJS
        module.exports = factory(require('jquery'));
    } else {
        // Global
        factory(jQuery);
    }
})(function($) {
    /*
    *  internal
    */

    var _previousResizeWidth = -1,
        _updateTimeout = -1;

    /*
    *  _parse
    *  value parse utility function
    */

    var _parse = function(value) {
        // parse value and convert NaN to 0
        return parseFloat(value) || 0;
    };

    /*
    *  _rows
    *  utility function returns array of jQuery selections representing each row
    *  (as displayed after float wrapping applied by browser)
    */

    var _rows = function(elements) {
        var tolerance = 1,
            $elements = $(elements),
            lastTop = null,
            rows = [];

        // group elements by their top position
        $elements.each(function(){
            var $that = $(this),
                top = $that.offset().top - _parse($that.css('margin-top')),
                lastRow = rows.length > 0 ? rows[rows.length - 1] : null;

            if (lastRow === null) {
                // first item on the row, so just push it
                rows.push($that);
            } else {
                // if the row top is the same, add to the row group
                if (Math.floor(Math.abs(lastTop - top)) <= tolerance) {
                    rows[rows.length - 1] = lastRow.add($that);
                } else {
                    // otherwise start a new row group
                    rows.push($that);
                }
            }

            // keep track of the last row top
            lastTop = top;
        });

        return rows;
    };

    /*
    *  _parseOptions
    *  handle plugin options
    */

    var _parseOptions = function(options) {
        var opts = {
            byRow: true,
            property: 'height',
            target: null,
            remove: false
        };

        if (typeof options === 'object') {
            return $.extend(opts, options);
        }

        if (typeof options === 'boolean') {
            opts.byRow = options;
        } else if (options === 'remove') {
            opts.remove = true;
        }

        return opts;
    };

    /*
    *  matchHeight
    *  plugin definition
    */

    var matchHeight = $.fn.matchHeight = function(options) {
        var opts = _parseOptions(options);

        // handle remove
        if (opts.remove) {
            var that = this;

            // remove fixed height from all selected elements
            this.css(opts.property, '');

            // remove selected elements from all groups
            $.each(matchHeight._groups, function(key, group) {
                group.elements = group.elements.not(that);
            });

            // TODO: cleanup empty groups

            return this;
        }

        if (this.length <= 1 && !opts.target) {
            return this;
        }

        // keep track of this group so we can re-apply later on load and resize events
        matchHeight._groups.push({
            elements: this,
            options: opts
        });

        // match each element's height to the tallest element in the selection
        matchHeight._apply(this, opts);

        return this;
    };

    /*
    *  plugin global options
    */

    matchHeight.version = '0.7.0';
    matchHeight._groups = [];
    matchHeight._throttle = 80;
    matchHeight._maintainScroll = false;
    matchHeight._beforeUpdate = null;
    matchHeight._afterUpdate = null;
    matchHeight._rows = _rows;
    matchHeight._parse = _parse;
    matchHeight._parseOptions = _parseOptions;

    /*
    *  matchHeight._apply
    *  apply matchHeight to given elements
    */

    matchHeight._apply = function(elements, options) {
        var opts = _parseOptions(options),
            $elements = $(elements),
            rows = [$elements];

        // take note of scroll position
        var scrollTop = $(window).scrollTop(),
            htmlHeight = $('html').outerHeight(true);

        // get hidden parents
        var $hiddenParents = $elements.parents().filter(':hidden');

        // cache the original inline style
        $hiddenParents.each(function() {
            var $that = $(this);
            $that.data('style-cache', $that.attr('style'));
        });

        // temporarily must force hidden parents visible
        $hiddenParents.css('display', 'block');

        // get rows if using byRow, otherwise assume one row
        if (opts.byRow && !opts.target) {

            // must first force an arbitrary equal height so floating elements break evenly
            $elements.each(function() {
                var $that = $(this),
                    display = $that.css('display');

                // temporarily force a usable display value
                if (display !== 'inline-block' && display !== 'flex' && display !== 'inline-flex') {
                    display = 'block';
                }

                // cache the original inline style
                $that.data('style-cache', $that.attr('style'));

                $that.css({
                    'display': display,
                    'padding-top': '0',
                    'padding-bottom': '0',
                    'margin-top': '0',
                    'margin-bottom': '0',
                    'border-top-width': '0',
                    'border-bottom-width': '0',
                    'height': '100px',
                    'overflow': 'hidden'
                });
            });

            // get the array of rows (based on element top position)
            rows = _rows($elements);

            // revert original inline styles
            $elements.each(function() {
                var $that = $(this);
                $that.attr('style', $that.data('style-cache') || '');
            });
        }

        $.each(rows, function(key, row) {
            var $row = $(row),
                targetHeight = 0;

            if (!opts.target) {
                // skip apply to rows with only one item
                if (opts.byRow && $row.length <= 1) {
                    $row.css(opts.property, '');
                    return;
                }

                // iterate the row and find the max height
                $row.each(function(){
                    var $that = $(this),
                        style = $that.attr('style'),
                        display = $that.css('display');

                    // temporarily force a usable display value
                    if (display !== 'inline-block' && display !== 'flex' && display !== 'inline-flex') {
                        display = 'block';
                    }

                    // ensure we get the correct actual height (and not a previously set height value)
                    var css = { 'display': display };
                    css[opts.property] = '';
                    $that.css(css);

                    // find the max height (including padding, but not margin)
                    if ($that.outerHeight(false) > targetHeight) {
                        targetHeight = $that.outerHeight(false);
                    }

                    // revert styles
                    if (style) {
                        $that.attr('style', style);
                    } else {
                        $that.css('display', '');
                    }
                });
            } else {
                // if target set, use the height of the target element
                targetHeight = opts.target.outerHeight(false);
            }

            // iterate the row and apply the height to all elements
            $row.each(function(){
                var $that = $(this),
                    verticalPadding = 0;

                // don't apply to a target
                if (opts.target && $that.is(opts.target)) {
                    return;
                }

                // handle padding and border correctly (required when not using border-box)
                if ($that.css('box-sizing') !== 'border-box') {
                    verticalPadding += _parse($that.css('border-top-width')) + _parse($that.css('border-bottom-width'));
                    verticalPadding += _parse($that.css('padding-top')) + _parse($that.css('padding-bottom'));
                }

                // set the height (accounting for padding and border)
                $that.css(opts.property, (targetHeight - verticalPadding) + 'px');
            });
        });

        // revert hidden parents
        $hiddenParents.each(function() {
            var $that = $(this);
            $that.attr('style', $that.data('style-cache') || null);
        });

        // restore scroll position if enabled
        if (matchHeight._maintainScroll) {
            $(window).scrollTop((scrollTop / htmlHeight) * $('html').outerHeight(true));
        }

        return this;
    };

    /*
    *  matchHeight._applyDataApi
    *  applies matchHeight to all elements with a data-match-height attribute
    */

    matchHeight._applyDataApi = function() {
        var groups = {};

        // generate groups by their groupId set by elements using data-match-height
        $('[data-match-height], [data-mh]').each(function() {
            var $this = $(this),
                groupId = $this.attr('data-mh') || $this.attr('data-match-height');

            if (groupId in groups) {
                groups[groupId] = groups[groupId].add($this);
            } else {
                groups[groupId] = $this;
            }
        });

        // apply matchHeight to each group
        $.each(groups, function() {
            this.matchHeight(true);
        });
    };

    /*
    *  matchHeight._update
    *  updates matchHeight on all current groups with their correct options
    */

    var _update = function(event) {
        if (matchHeight._beforeUpdate) {
            matchHeight._beforeUpdate(event, matchHeight._groups);
        }

        $.each(matchHeight._groups, function() {
            matchHeight._apply(this.elements, this.options);
        });

        if (matchHeight._afterUpdate) {
            matchHeight._afterUpdate(event, matchHeight._groups);
        }
    };

    matchHeight._update = function(throttle, event) {
        // prevent update if fired from a resize event
        // where the viewport width hasn't actually changed
        // fixes an event looping bug in IE8
        if (event && event.type === 'resize') {
            var windowWidth = $(window).width();
            if (windowWidth === _previousResizeWidth) {
                return;
            }
            _previousResizeWidth = windowWidth;
        }

        // throttle updates
        if (!throttle) {
            _update(event);
        } else if (_updateTimeout === -1) {
            _updateTimeout = setTimeout(function() {
                _update(event);
                _updateTimeout = -1;
            }, matchHeight._throttle);
        }
    };

    /*
    *  bind events
    */

    // apply on DOM ready event
    $(matchHeight._applyDataApi);

    // update heights on load and resize events
    $(window).bind('load', function(event) {
        matchHeight._update(false, event);
    });

    // throttled update heights on resize events
    $(window).bind('resize orientationchange', function(event) {
        matchHeight._update(true, event);
    });

});

jQuery(document).ready( function($) { 
	jQuery('.lgc-equal-heights').matchHeight(); 
	
	var rows = $.fn.matchHeight._rows($('.lgc-equal-heights'));

	$.each(rows, function(i, row) {
		row.first().addClass('lgc-first');
		row.last().addClass('lgc-last');
	});
	$('.lgc-equal-heights').on('click', function(){
		var _this = $( this );
		var height = _this.children().height();
		
		$( this ).on('transitionend webkitTransitionEnd oTransitionEnd otransitionend MSTransitionEnd', function( event ) {
			var changedHeight = _this.children().height();
			if ( height !== changedHeight ) {
				$.fn.matchHeight._update();
				$.fn.matchHeight._maintainScroll = true;
			}
		});
	});
});

js/jquery.matchHeight-min.js
!function(a){"use strict";"function"==typeof define&&define.amd?define(["jquery"],a):"undefined"!=typeof module&&module.exports?module.exports=a(require("jquery")):a(jQuery)}(function(a){var b=-1,c=-1,d=function(a){return parseFloat(a)||0},e=function(b){var c=1,e=a(b),f=null,g=[];return e.each(function(){var b=a(this),e=b.offset().top-d(b.css("margin-top")),h=g.length>0?g[g.length-1]:null;null===h?g.push(b):Math.floor(Math.abs(f-e))<=c?g[g.length-1]=h.add(b):g.push(b),f=e}),g},f=function(b){var c={byRow:!0,property:"height",target:null,remove:!1};return"object"==typeof b?a.extend(c,b):("boolean"==typeof b?c.byRow=b:"remove"===b&&(c.remove=!0),c)},g=a.fn.matchHeight=function(b){var c=f(b);if(c.remove){var d=this;return this.css(c.property,""),a.each(g._groups,function(a,b){b.elements=b.elements.not(d)}),this}return this.length<=1&&!c.target?this:(g._groups.push({elements:this,options:c}),g._apply(this,c),this)};g.version="0.7.0",g._groups=[],g._throttle=80,g._maintainScroll=!1,g._beforeUpdate=null,g._afterUpdate=null,g._rows=e,g._parse=d,g._parseOptions=f,g._apply=function(b,c){var h=f(c),i=a(b),j=[i],k=a(window).scrollTop(),l=a("html").outerHeight(!0),m=i.parents().filter(":hidden");return m.each(function(){var b=a(this);b.data("style-cache",b.attr("style"))}),m.css("display","block"),h.byRow&&!h.target&&(i.each(function(){var b=a(this),c=b.css("display");"inline-block"!==c&&"flex"!==c&&"inline-flex"!==c&&(c="block"),b.data("style-cache",b.attr("style")),b.css({display:c,"padding-top":"0","padding-bottom":"0","margin-top":"0","margin-bottom":"0","border-top-width":"0","border-bottom-width":"0",height:"100px",overflow:"hidden"})}),j=e(i),i.each(function(){var b=a(this);b.attr("style",b.data("style-cache")||"")})),a.each(j,function(b,c){var e=a(c),f=0;if(h.target)f=h.target.outerHeight(!1);else{if(h.byRow&&e.length<=1)return void e.css(h.property,"");e.each(function(){var b=a(this),c=b.attr("style"),d=b.css("display");"inline-block"!==d&&"flex"!==d&&"inline-flex"!==d&&(d="block");var e={display:d};e[h.property]="",b.css(e),b.outerHeight(!1)>f&&(f=b.outerHeight(!1)),c?b.attr("style",c):b.css("display","")})}e.each(function(){var b=a(this),c=0;h.target&&b.is(h.target)||("border-box"!==b.css("box-sizing")&&(c+=d(b.css("border-top-width"))+d(b.css("border-bottom-width")),c+=d(b.css("padding-top"))+d(b.css("padding-bottom"))),b.css(h.property,f-c+"px"))})}),m.each(function(){var b=a(this);b.attr("style",b.data("style-cache")||null)}),g._maintainScroll&&a(window).scrollTop(k/l*a("html").outerHeight(!0)),this},g._applyDataApi=function(){var b={};a("[data-match-height], [data-mh]").each(function(){var c=a(this),d=c.attr("data-mh")||c.attr("data-match-height");d in b?b[d]=b[d].add(c):b[d]=c}),a.each(b,function(){this.matchHeight(!0)})};var h=function(b){g._beforeUpdate&&g._beforeUpdate(b,g._groups),a.each(g._groups,function(){g._apply(this.elements,this.options)}),g._afterUpdate&&g._afterUpdate(b,g._groups)};g._update=function(d,e){if(e&&"resize"===e.type){var f=a(window).width();if(f===b)return;b=f}d?c===-1&&(c=setTimeout(function(){h(e),c=-1},g._throttle)):h(e)},a(g._applyDataApi),a(window).bind("load",function(a){g._update(!1,a)}),a(window).bind("resize orientationchange",function(a){g._update(!0,a)})}),jQuery(document).ready(function(a){jQuery(".lgc-equal-heights").matchHeight();var b=a.fn.matchHeight._rows(a(".lgc-equal-heights"));a.each(b,function(a,b){b.first().addClass("lgc-first"),b.last().addClass("lgc-last")}),a(".lgc-equal-heights").on("click",function(){var b=a(this),c=b.children().height();a(this).on("transitionend webkitTransitionEnd oTransitionEnd otransitionend MSTransitionEnd",function(d){var e=b.children().height();c!==e&&(a.fn.matchHeight._update(),a.fn.matchHeight._maintainScroll=!0)})})});

lightweight-grid-columns.php
<?php
/*
Plugin Name: Lightweight Grid Columns
Plugin URI: http://generatepress.com
Description: Add columns to your content using easy to use shortcodes.
Version: 0.6
Author: Thomas Usborne
Author URI: https://tomusborne.com
License: GNU General Public License v2 or later
License URI: http://www.gnu.org/licenses/gpl-2.0.html
Text Domain: lightweight-grid-columns
*/

define( 'LGC_VERSION', '0.6');

/**
 * Load plugin textdomain.
 *
 * @since 0.1
 */
add_action( 'plugins_loaded', 'lgc_load_textdomain' );
function lgc_load_textdomain() {
	load_plugin_textdomain( 'lightweight-grid-columns' ); 
}

if ( ! function_exists( 'lgc_shortcodes_register_shortcode' ) ) :
/*
 * Declare our shortcode
 */
add_action( 'init','lgc_shortcodes_register_shortcode' );
function lgc_shortcodes_register_shortcode()
{
	add_shortcode( 'lgc_column', 'lgc_columns_shortcode' );
}
endif;

if ( ! function_exists( 'lgc_add_shortcode_button' ) ) :
/*
 * Set it up so we can register our TinyMCE button
 */
add_action('admin_init', 'lgc_add_shortcode_button');
function lgc_add_shortcode_button() 
{
	// check user permissions
	if ( !current_user_can( 'edit_posts' ) && !current_user_can( 'edit_pages' ) )
		return;
		
	// check if WYSIWYG is enabled
	if ( get_user_option( 'rich_editing' ) == 'true') {
		add_filter( 'mce_external_plugins', 'lgc_shortcodes_add_tinymce_plugin' );
		add_filter( 'mce_buttons', 'lgc_shortcodes_register_button' );
	}
}
endif;

if ( ! function_exists( 'lgc_shortcodes_add_tinymce_plugin' ) ) :
/*
 * Register our tinyMCE button javascript
 */
function lgc_shortcodes_add_tinymce_plugin( $plugin_array ) {
	$plugin_array[ 'lgc_shortcodes_button' ] = plugins_url( '/js/button.js', __FILE__ );
	return $plugin_array;
}
endif;

if ( ! function_exists( 'lgc_shortcodes_register_button' ) ) :
/*
 * Register our TinyMCE button
 */
function lgc_shortcodes_register_button( $buttons ) {
	array_push( $buttons, 'lgc_shortcodes_button' );
	return $buttons;
}
endif;

if ( ! function_exists( 'lgc_translatable_strings' ) ) :
add_action( 'admin_head','lgc_translatable_strings', 0 );
function lgc_translatable_strings()
{
	?>
	<script type="text/javascript">
		var lgc_add_columns = '<?php _e( 'Add columns', 'lightweight-grid-columns' ); ?>';
		var lgc_columns = '<?php _e( 'Columns', 'lightweight-grid-columns' ); ?>';
		var lgc_desktop = '<?php _e( 'Desktop grid percentage', 'lightweight-grid-columns' ); ?>';
		var lgc_tablet = '<?php _e( 'Tablet grid percentage', 'lightweight-grid-columns' ); ?>';
		var lgc_mobile = '<?php _e( 'Mobile grid percentage', 'lightweight-grid-columns' ); ?>';
		var lgc_content = '<?php _e( 'Content', 'lightweight-grid-columns' ); ?>';
		var lgc_last = '<?php _e( 'Last column in row?', 'lightweight-grid-columns' ); ?>';
	</script>
	<?php
}
endif;

if ( ! function_exists( 'lgc_shortcodes_admin_css' ) ) :
/*
 * Add our admin CSS
 */
add_action( 'admin_enqueue_scripts', 'lgc_shortcodes_admin_css' );
function lgc_shortcodes_admin_css() {
	wp_enqueue_style( 'lgc-columns-admin', plugins_url('/css/admin.css', __FILE__) );
}
endif;

if ( ! function_exists( 'lgc_shortcodes_css' ) ) :
/*
 * Add the unsemantic framework
 */
add_action( 'wp_enqueue_scripts', 'lgc_shortcodes_css', 99 );
function lgc_shortcodes_css() {
	wp_enqueue_style( 'lgc-unsemantic-grid-responsive-tablet', plugins_url('/css/unsemantic-grid-responsive-tablet.css', __FILE__), array(), LGC_VERSION, 'all' );
	wp_register_script( 'lgc-matchHeight', plugins_url('/js/jquery.matchHeight-min.js', __FILE__), array( 'jquery' ), LGC_VERSION, true );
}
endif;

if ( ! function_exists( 'lgc_columns_shortcode' ) ) :
/*
 * Create the output of the columns shortcode
 */
function lgc_columns_shortcode( $atts , $content = null ) {

	extract( shortcode_atts(
		array(
			'grid' => '50',
			'tablet_grid' => '50',
			'mobile_grid' => '100',
			'last' => '',
			'class' => '',
			'style' => '',
			'equal_heights' => 'true',
			'id' => ''
		), $atts )
	);
	
	if ( 'true' == $equal_heights )
		wp_enqueue_script( 'lgc-matchHeight' );
	
	$content = sprintf( 
		'<div %9$s class="lgc-column lgc-grid-parent %1$s %2$s %3$s %4$s"><div %6$s class="inside-grid-column %5$s">%7$s</div></div>%8$s',
		'lgc-grid-' . intval( $grid ),
		'lgc-tablet-grid-' . intval( $tablet_grid ),
		'lgc-mobile-grid-' . intval( $mobile_grid ),
		( 'true' == $equal_heights ) ? 'lgc-equal-heights' : '',
		esc_attr( $class ),
		( '' !== $style ) ? ' style="' . esc_attr( $style ) . '"' : '',
		do_shortcode( $content ),
		( 'true' == $last ) ? '<div class="lgc-clear"></div>' : '',
		( '' !== $id ) ? 'id="' . $id . '"' : ''
	);
	
	return force_balance_tags( $content );

}
endif;

if ( ! function_exists( 'lgc_columns_helper' ) ) :
/*
 * Fix the WP paragraph and <br /> issue with shortcodes
 */
add_filter( 'the_content', 'lgc_columns_helper' );
function lgc_columns_helper( $content ) 
{
    $array = array (
        '<p>[lgc_column' => '[lgc_column', 
        'lgc_column]</p>' => 'lgc_column]', 
		'<br />[lgc_column' => '[lgc_column',
        'lgc_column]<br />' => 'lgc_column]'
    );

    return strtr( $content, $array ); 
}
endif;

if ( ! function_exists( 'lgc_ie_compatibility' ) ) :
/** 
 * Add compatibility for IE8 and lower
 * @since 0.3
 */
add_action('wp_head','lgc_ie_compatibility');
function lgc_ie_compatibility()
{
?>
	<!--[if lt IE 9]>
		<link rel="stylesheet" href="<?php echo plugins_url('/css/ie.min.css', __FILE__); ?>" />
	<![endif]-->
<?php
}
endif;